pkg/controllers/management/auth/project_cluster_handler.go

```diff
@@ -91,6 +96,29 @@ func (l *projectLifecycle) sync(key string, orig *v3.Project) error {
 		}
 	}
 
+	// create cluster role for this cluster's templates and templateversions. cluster role here too since we want to grant access to templates resource
+	// which is not namespaced
+	clusterName := orig.Namespace
+	gotCR, err := l.mgr.clusterRoleClient.Get("project-"+clusterName+"-"+orig.Name+"-use-templates-templateversions", metav1.GetOptions{})
+
+	newCR := &rbacv1.ClusterRole{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "project-" + clusterName + "-" + orig.Name + "-use-templates-templateversions",
+		},
+	}
+
+	// create cluster role only if it doesn't already exist
+	if !reflect.DeepEqual(gotCR, newCR) {
```

Comment:
we should only create when a) line 102 err != nil && err == IsAlreadyExists OR gotCR is nil. This way we save on deepEquals code